src/Security/SecIdentity2.cs

```diff
@@ -0,0 +1,84 @@
+//
+// SecIdentity2.cs: Bindings the Security's sec_identity_t
+//
+// The difference between SecIdentity2 and SecIdentity is that the
+// SecIdentity2 is a binding for the new sec_identity_t API that was
+// introduced on iOS 12/OSX Mojave, while SecIdentity is the older API
+// that binds SecIdentityRef.
+//
+// Authors:
+//   Miguel de Icaza (miguel@microsoft.com)
+//
+// Copyrigh 2018 Microsoft Inc
+//
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+using ObjCRuntime;
+using Foundation;
+using CoreFoundation;
+
+namespace Security {
+	
+	[TV (12,0), Mac (10,14), iOS (12,0)]
+	public class SecIdentity2 : NativeObject {
+		internal SecIdentity2 (IntPtr handle) : base (handle, false) {} 
+		public SecIdentity2 (IntPtr handle, bool owns) : base (handle, owns) {}
+
+#if !COREBUILD
+		[TV (12,0), Mac (10,14), iOS (12,0)]
+		[DllImport (Constants.SecurityLibrary)]
+		extern static IntPtr sec_identity_create (IntPtr secidentityHandle);
+
+		[TV (12,0), Mac (10,14), iOS (12,0)]
+		public SecIdentity2 (SecIdentity identity)
+		{
+			if (identity == null)
+				throw new ArgumentNullException (nameof (identity));
+			
+			handle = sec_identity_create (identity == null ? IntPtr.Zero : identity.Handle);
```

Comment:
That API can return `null` so `InitializeHandle` (from base `NativeObject`) should be called to avoid creating an invalid instance (that will crash/throw later).

Reply:
Got it - will use the property version.